### ISSUE TITLE
NoMethodError in intercom/request 

### ISSUE THREAD

**Issue by jvenezia** [intercom/intercom-ruby#339]
## Version info
  - intercom-ruby version: **3.5.12**
  - Ruby version: **2.3.1**

## Expected behavior
Comprehensive exception like `Intercom::ServiceUnavailableError` or anything.

## Actual behavior
got `NoMethodError` exception.

```
NoMethodError: undefined method `[]' for nil:NilClass
  from intercom/request.rb:94:in `parse_body'
  from intercom/request.rb:70:in `block in execute'
  from net/http.rb:853:in `start'
  from intercom/request.rb:65:in `execute'
  from intercom/client.rb:111:in `execute_request'
  from intercom/client.rb:88:in `get'
  from intercom/api_operations/find.rb:13:in `find'
```

**Comment by choran** [comment]
Hi @jvenezia,
Can you provide a little more context? Was there a specific method you were trying that was failing and you think the error was unclear for that? Or was it related to a more general issue with error handling? Just wasnt sure which one it was.
Thanks
Cathal

**Comment by jvenezia** [comment]
The problem is that I have a `NoMethodError` coming from the Intercom gem, and I have no control on it. It looks like a bug to me.

**Comment by choran** [comment]
Yep, makes sense.
It seems like we should be returning nil there which is what is being returned from the json parse in the parse_body method.
Not sure if nil being returned would make sense in your particular context but, like you said, at least it would give you control over it.

**Comment by choran** [comment]
@ipc103 has added a fix for this 👍 in pr #353 
Just testing this now ... it looks good so will merge this in shortly
thanks
